scipy/optimize/_lbfgsb_py.py

```diff
@@ -360,9 +362,9 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
         elif task_str.startswith(b'NEW_X'):
             # new iteration
             n_iterations += 1
-            if callback is not None:
-                callback(np.copy(x))
 
+            if _call_callback(callback, x, f):
+                task[:] = 'STOP: CALLBACK SAYS SO!'
```

Comment:
`STOP: CALLBACK REQUESTED HALT`?

Reply:
This won't show up anywhere; it gets replaced. I was just having fun : P